### ISSUE TITLE
question on loss function

### ISSUE THREAD

**Issue by Weekday34** [Shengyu-Feng/RLD4CO#2]
Hello, I was interested in approximating probability distribution for CO and read your paper interestingly.

During skimming your code, I got a question.

1.
at line 267, in solver.py, you use reward term: original energy - energy as Hamiltonian term.

In this point, what is the advantage of using original energy in addition to energy?

In my knowledge, I guess it is sufficient to use energy term as Hamiltonian term for loss, but I'm not sure.

Is there any meaningful effect on the "original energy" term?

2.
Plus, I wonder why this paper use flipping probability, q(x'_i = 1-x_i | x) instead of q(x'_i = 1 | x).

Actually, some papers use q(x'_i = 1 | x) and other papers use q(x'_i = 1-x_i | x).

Does it depends on the researcher's preference?

**Comment by Shengyu-Feng** [comment]
Hi, thank you for your questions and apologize the late response. My GitHub email account is an old one, you can reach me through my current email address listed in README.md

For the first question, there is no difference after normalization, here I simply follow the formula of Langevin dynamics. 

For the second question, it is about which target is easier to learn. The flipping probability is in general a linear one for QUBO as you can see from RLSA. But I think neural networks may make these two learning targets no big difference, after over-parameterization. 

But it is better to have more careful ablation studies to jump into the conclusions here.

**Comment by Weekday34** [comment]
Thank you for your replies.
